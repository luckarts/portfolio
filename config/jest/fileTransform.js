const path = require('path');

// This is a custom Jest transformer turning file imports into filenames.
// https://jest-bot.github.io/jest/docs/webpack.html

module.exports = {
  process(src, filename) {
    return `module.exports = ${JSON.stringify(path.basename(filename))};`;
  }
};