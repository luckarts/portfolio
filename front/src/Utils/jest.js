import checkPropTypes from "check-prop-types";
import { createStore } from "redux";
import rootReducer from "store/reducer";

export const checkProps = (component, expectedProps) => {
  // @ts-ignore
  const ErrProps = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return ErrProps;
};
export const testStore = initialState => {
  return createStore(rootReducer, initialState);
};
