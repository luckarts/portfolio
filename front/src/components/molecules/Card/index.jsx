import React from 'react';
import PropTypes from 'prop-types';
import { LazyImage } from 'lucky-react-component';

const propTypes = {
	className: PropTypes.string,
	i: PropTypes.number.isRequired,
	photo: PropTypes.string,
};

const Card = ({ className, photo, i }) => {
	return <div className={className}>{photo && <LazyImage key={i} src={'/img/' + photo} alt={photo} placeholder={'/thumb/img/' + photo} />}</div>;
};
Card.propTypes = propTypes;
export default Card;
