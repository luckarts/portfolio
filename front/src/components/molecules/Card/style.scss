//card Article
.lazy-load-image-background {
    display: block !important;
}

.card {
    border-right: 1px solid rgb(240, 237, 237);
    border-radius: 10px;
    padding: 0px;
    box-shadow: 0px 0px 11px 0px rgba(1, 67, 110, 0.1);
    border-color: rgb(238, 238, 238);

    &:hover {
        box-shadow: 0px 0px 11px 0px rgba(1, 67, 110, 0.3);

    }

    &.hover {
        transform-origin: center;
        transition: transform .4s;
        transform: scale(1.05);
    }


    &-header {
        background-size: cover;
        background-position: center;
        background-color: var(--color-contrast);
        border-radius: 10px 10px 0 0;
        height: 300px;
        width: 100%;
        position: relative;

        @media (max-width: 500px) {
            height: 250px;
        }

        img {
            width: 100%;
            margin: auto;
            text-align: center;
            border-radius: 10px;

            &:hover {
                box-shadow: 10px 10px 0 0 rgba(43, 10, 61, 0.2);
                opacity: 0.5;
            }
        }
    }

    &-content {
        padding: 15px;
        background-color: var(--color-contrast);
        border-radius: 0 0 10px 10px;
    }

    &-project {

        width: 100%;
        opacity: 0;
        animation-name: slideInRight;
        animation-duration: 1s;
        transition-timing-function: ease;
        animation-fill-mode: forwards;



        &:hover {
            .figcaption {
                display: flex !important;
            }

        }

        @media (max-width: 900px) {
            .figcaption {
                display: flex !important;
            }
        }

    }
}

//card profile

.card {
    &-inner {
        opacity: 0;
        animation-name: slideInRight;
        animation-duration: 1s;
        transition-timing-function: ease;
        animation-fill-mode: forwards;
        animation-delay: 0.5;

        margin: auto;

    }

    &-profile {
        padding: 15px;
        margin-top: 5vw;
        background-color: var(--color-contrast);
        border-radius: 10px;
        text-align: center;

        h2 {
            padding-top: 1rem;
        }

        .row {
            justify-content: space-around;
        }

        h2 {

            font-size: calc(40px + (26 - 14) * ((100vw - 300px) / (1600 - 300)));
            width: 100%;

            font-weight: bold;
            text-align: center;
            border: 0;
        }

        p {
            padding-bottom: 5px;
            max-width: 500px;
            margin-left: auto;
            margin-right: auto;
            border-bottom: 1px solid var(--color-primary)
        }

        .icon-profile {
            max-width: 60%;

            @media screen and (max-width: 640px) {
                max-width: 50%;
            }
        }

    }

    .icon-profile {
        display: flex;
        justify-content: center;
        border-radius: 300px;
        max-height: 200px;
    }

    .reseaux {

        @extend .row;
        padding-top: 1rem;
        justify-content: center;
        align-items: center;

    }
}

//card masonry
.card-masonry {
    margin: 1vw 0.5vw;
    transform: scale(1);
    transition: ease-in-out transform .5s;

    img {

        border-radius: 10px;
        width: 100%;
        transform: scale(1.0);
        transition: ease-in-out transform .5s;
        object-fit: fill;

        &:hover {
            transform: scale(1.05);
            transition: ease-in-out transform .5s;
        }

    }



}

.experiences {
    background-color: var(--color-contrast);

    @media screen and (min-width: 980px) {
        &.after-left {
            position: relative;

            &::after {
                position: absolute;
                top: 27px;
                right: -14px;
                display: inline-block;
                border-top: 14px solid transparent;
                border-left: 14px solid var(--color-contrast);
                border-right: 0 solid var(--color-contrast);
                border-bottom: 14px solid transparent;

                content: " ";
            }
        }

        &.after-right {
            position: relative;

            &::after {
                position: absolute;
                bottom: 20px;
                left: -14px;
                display: inline-block;
                border-top: 14px solid transparent;
                border-right: 14px solid var(--color-contrast);
                border-left: 0 solid var(--color-contrast);
                border-bottom: 14px solid transparent;

                content: " ";
            }
        }
    }




    h4 {
        font-size: 1.5em;
        margin: 0;
        color: var(--color-primary)
    }

    li {
        padding: 5px 0;
        font-size: 0.9rem;
        color: var(--color-primary)
    }

    ul {
        padding-left: 15px;
    }
}

.image-gallery-left-nav .image-gallery-svg,
.image-gallery-right-nav .image-gallery-svg {
    display: none;
}

.image-gallery-fullscreen-button {
    display: none;
}