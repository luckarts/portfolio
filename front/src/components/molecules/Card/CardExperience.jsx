import React from 'react';
import parse from 'html-react-parser';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './style.scss';

const propTypes = {
	className: PropTypes.string.isRequired,
	experience: PropTypes.object,
};

const CardExperience = ({ className, edit, date, fonction, link, company, list_experience }) => {
	const component = (
		<div className={'card ' + className}>
			<p>{date}</p>
			<h4> {fonction}</h4>
			<a href={link}>
				<h3> {company}</h3>
			</a>

			{list_experience.length > 0 && (
				<ul>
					{list_experience.map((listExp, index) => (
						<li key={index}>{parse(listExp.description)}</li>
					))}
				</ul>
			)}
		</div>
	);

	return edit ? <Link to={`/edit/experience/${company}`}> {component}</Link> : component;
};
CardExperience.propTypes = propTypes;
export default CardExperience;
