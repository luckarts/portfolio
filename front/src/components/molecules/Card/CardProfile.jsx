import React from 'react';
import './style.scss';
import PropTypes from 'prop-types';

const propTypes = {
	user: PropTypes.object,
};

const CardProject = ({ cv }) => {
	return (
		<div className="card-profile card">
			<div className="row">
				<img className="icon-profile" src={'/img/character.png'} alt="img" />

				<div className=" col-lg-7 col-sm-12">
					<h2>Bachelerie Luc</h2>
					<p>
						DÃ©veloppeur Front end Javascript <br />
					</p>
					<div className="reseaux">
						<a className="link-icon" href={'/upload/' + cv} target="_self" rel="noopener noreferrer">
							<img src="/img/logocv.jpg" alt="cv" />
						</a>
						<a className="link-icon" href="https://www.linkedin.com/in/luc-bachelerie-developperjs/" rel="noopener noreferrer">
							<img src="/img/linkedlin.jpg" alt="linkedlin" />
						</a>
						<a className="link-icon" href="mailto:luc.bachelerieart@gmail.com">
							<img src="/img/email.jpg" alt="email" />
						</a>
						<a className="link-icon" href="tel:0771073054">
							<img src="/img/tel.jpg" alt="tel" />
						</a>
					</div>
				</div>
			</div>
		</div>
	);
};
CardProject.propTypes = propTypes;
export default CardProject;
