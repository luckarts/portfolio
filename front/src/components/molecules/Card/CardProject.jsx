import React from 'react';
import './style.scss';
import parse from 'html-react-parser';
import PropTypes from 'prop-types';
import { Button } from '../../atoms/Button';
const propTypes = {
	title: PropTypes.string,
	img: PropTypes.string,
	techno: PropTypes.string,
	description: PropTypes.string,
	linkCode: PropTypes.string,
	linkWebsite: PropTypes.string,
	edit: PropTypes.bool,
	id: PropTypes.number,
};

const CardProject = ({ title, img, techno, description, linkCode, linkWebsite, edit, id }) => {
	let link;
	if (edit) {
		let slug = title.toLowerCase();
		link = `/edit/project/${slug}`;
	}

	return (
		<div className="card card-project">
			<div className="card-header" style={{ backgroundImage: 'url(' + img + ')' }}>
				<div className="figcaption">
					{linkCode && (
						<Button color="secondary" variante="solid" rounded="rounded" href={linkCode}>
							Code
						</Button>
					)}
					{linkWebsite && (
						<Button color="secondary" variante="solid" rounded="rounded" href={linkWebsite}>
							Demo
						</Button>
					)}

					{edit && (
						<Button color="secondary" variante="solid" rounded="rounded" link={link}>
							Edit
						</Button>
					)}
				</div>
			</div>

			<div className="card-content">
				<h3 className="card-title">{title}</h3>
				{description && <p>{parse(description)}</p>}

				<p>
					Techno: <span className="bold">{techno}</span>
				</p>
			</div>
		</div>
	);
};
CardProject.propTypes = propTypes;
export default CardProject;
