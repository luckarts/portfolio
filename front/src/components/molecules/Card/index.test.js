import React from "react";
import { shallow, mount } from "enzyme";
import Card from "./index";
import { checkProps } from "Utils/jest";

let wrapper, props;
let openLightbox = jest.fn();
props = {
  className: "test",
  i: 1,
  photos: "test",
  openLightbox: () => openLightbox(1)
};

it("Should render.", () => {
  expect(mount(<Card {...props} />)).toMatchSnapshot();
});

describe("Galerie Component with children props", () => {
  beforeEach(() => {
    wrapper = shallow(<Card {...props} />);
    wrapper.find("img").simulate("click");
  });

  it(" Checking PropTypes no error", () => {
    const PropsError = checkProps(Card, props);
    expect(PropsError).toBeUndefined();
  });
  it('Should call "openLightbox" when clicking on it', () => {
    expect(openLightbox).toHaveBeenCalled();
  });
});
