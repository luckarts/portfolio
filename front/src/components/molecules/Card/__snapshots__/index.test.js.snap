// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Should render. 1`] = `ReactWrapper {}`;
